framework/source/class/qx/ui/form/core/VirtualDropDownList.js

```diff
@@ -439,19 +439,24 @@ qx.Class.define("qx.ui.form.core.VirtualDropDownList",
     __adjustHeight : function()
     {
       var availableHeigth = this.__getAvailableHeigth();
-      var maxListHeight = this._target.getMaxListHeight();
+      var maxHeight = this._target.getMaxListHeight();
       var list = this.getChildControl("list");
       var itemsHeight = list.getPane().getRowConfig().getTotalSize();
 
-      if (maxListHeight == null || itemsHeight < maxListHeight) {
-        maxListHeight = itemsHeight;
+      if (maxHeight == null || itemsHeight < maxHeight) {
+        maxHeight = itemsHeight;
       }
 
-      if (maxListHeight > availableHeigth) {
-        list.setMaxHeight(availableHeigth);
-      } else if (maxListHeight < availableHeigth) {
-        list.setMaxHeight(maxListHeight);
+      if (maxHeight > availableHeigth) {
+        maxHeight = availableHeigth;
       }
+
+      var minHeight = list.getMinHeight();
+      if (null !== minHeight && minHeight > maxHeight) {
+        maxHeight = minHeight;
```

Comment:
Has this been tested at the extremes of container boundaries? Should there be tests of whether the increased max height will still fit in the container?

Reply:
I would argue that this is a non-issue because having maxHeight never be smaller than minHeight would be more important than whether or not it will fit in a container. I also don't think containers will be 42 pixels or less small.

Reply:
I don't understand where the 42 pixels come from, but the argument that `minHeight` should never be greater than `maxHeight` seems reasonable to me.  

WRT whether `minHeight` should also be checked against the container (ie against `availableHeight`), this is only an issue if one of `maxHeight`/`minHeight` is an invalid value so it seems a bit of a moot point, except to say that adjusting `maxHeight` up at least makes sure that it is visible and the developer can look further (or maybe the user can resize the viewport).

Reply:
Excuse me for not explaining where I got the 42 from.

When the `qx.ui.list.List` gets constructed, its base constructor `qx.ui.virtual.core.Scroller` gets called which in turn calls the base constructor of `qx.ui.core.scroll.AbstractScrollArea`. In this constructor a `size` gets calculated as follows:

```
var size = qx.ui.core.scroll.AbstractScrollArea.DEFAULT_SCROLLBAR_WIDTH * 2 + 14;
```

which will result in the number `42`.

This value will then be set as `minHeight` and `minWidth` of the List (LayoutItem).